apps/keyboard/build/layout_configurator.js

```diff
@@ -70,34 +81,31 @@ KeyboardLayoutDetail.prototype.load = function(appDir) {
     }
   }
 
-  // Handle its dictionary declarations, if any.
+  // Handle its dictionary declarations, and count the size, if any.
+  var lang;
   switch (this.imEngineId) {
     case 'latin':
       this.preloadDictRequired = false;
-      this.dictId = win.Keyboards[id].autoCorrectLanguage;
-      if (this.dictId) {
-        this.dictFilePath = 'dictionaries/' + this.dictId + '.dict';
-
-        var dictMetadata = JSON.parse(utils.getFileContent(
-          utils.getFile(appDir.path, 'js', 'imes', 'latin',
-            'dictionaries', 'dict_metadata.json')));
-
-        if (!dictMetadata[this.dictId] || !dictMetadata[this.dictId].label) {
-          throw new Error('KeyboardLayoutDetail: ' +
-            'Keyboard layout ' + id + '.js' +
-            ' specified a dictionary ' + this.dictId + ' without label ' +
-            'for latin engine.');
-        }
-        this.dictLabel = dictMetadata[this.dictId].label;
+      lang = win.Keyboards[id].autoCorrectLanguage;
+      if (lang) {
+        this.dictFilename = lang + '.dict';
+        this.dictFilePath = 'dictionaries/' + lang + '.dict';
         this.dictFile = utils.getFile(appDir.path, 'js', 'imes', 'latin',
-                                      'dictionaries', this.dictId + '.dict');
+                                      'dictionaries', lang + '.dict');
```

Comment:
(nit) Let's use ``this.dictFilename`` instead of ``lang + '.dict'`` since you already have it.